setup.py

```diff
@@ -96,6 +96,8 @@ def main():
 
         'license': 'MIT',
 
+        'python_requires': '>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, !=3.5.*',
```

Comment:
As I understand it, this "python_requires" statement would prevent pip (or at least recent versions of pip) from installing pyodbc on Python 3.5, 3.4, etc.  Do we really want to go that far?  It would be unusual (and not recommended) to install pyodbc on Python 3.4 but do we want to positively forbid it?

Reply:
Bump major version then?

Reply:
`python_requires` doesn't positively forbid installation, people can tell pip to ignore it.

```
  --ignore-requires-python    Ignore the Requires-Python information.
```

`python_requires` gives protection so you can use modern syntax without worrying you're going to break things for end users. And if you're not testing a version, you shouldn't give the impression it's supported.

On the whole, it's not just worth having half-in/half-out support, especially as old versions are so little used.

On the topic: we could also upgrade to modern Python syntax now (f-strings!). I could do it here or in a followup PR?